### PR TITLE
Marines can no longer violate the laws of physics through pure oorah

### LINKED ISSUE #490: you can leap over barricades into windows

### PR DESCRIPTION
**Pull request by F-OS**
fixes #490

### CODE REVIEW

**Comment by F-OS** [comment]
HELP

**Comment by F-OS** [comment]
WHAT THE FUCK DID I DO SAVE ME

File: code/game/objects/structures.dm [hunk 1]
```diff
@@ -119,9 +119,10 @@
 			return
 		if(user.loc == loc)
 			var/turf/target = get_step(T, dir)
-			if(target.density) //Turf is dense, not gonna work
-				to_chat(user, "<span class='warning'>You cannot leap this way.</span>")
-				return
+			for(var/atom/A in target.contents)
+   				if(A.density)
+					to_chat(user, "<span class='warning'>You cannot leap this way.</span>")
+					return
```
Comment (Ghommie) [inline code/game/objects/structures.dm]: Don't remove turf density check or players will "oorah" through wall turfs. Plus this overrides the control below `for(var/atom/movable/A in target)`, which also checks if the structure can be effectively climbed (I.E. window frame, tables).
On the aforementioned control, change the `var/` from `atom/movable/` to `obj/`, and find another way to fix the edge cases.

**Comment by Ghommie** [comment]
I found the source of the problems and am already working on a PR for it, and it's not about barricades.
Basically fulltile and framed windows were still possessing the ON_BORDER flag.

**Comment by F-OS** [comment]
@Ghommie Thanks, this PR was pissing me off because of the weird errors, have a good one!

**Pull request closed without merging**